### DISCUSSION TITLE
Building a "key" generator

### DISCUSSION THREAD

**Discussion by thehowl** [Budibase/budibase#5220] [Help]
Hello - thank you for the amazing product! I've been testing it out and it seems very good and definitely a solution to try to eventually remove some of the headaches of my life.

After building a small CRUD interface for a resource, I thought about moving on to the next step. Namely I wanted to have on a resource (ie. a client) a way to create from there N number of "access keys", which could then be distributed to access the application on its real frontend.

The way I tried to set this up was by adding this page which had this part:

![image](https://user-images.githubusercontent.com/4681308/161338845-933ccaf1-ce4e-4e27-9829-672b82c12f12.png)

and with the final goal of having "Create" create the number of keys specified in the form field.

Now the way that I'd expect the product to let me do it would be of 1. generating a SQL query from some JavaScript that I'd write, taking as a parameter said number of tokens and 2. execute the SQL query in the postgres database I'm using 3. print (or even better, list on their Activation Tokens page) said tokens.

Sadly, it seems that after tinkering with Budibase a bit, it doesn't seem exactly straightforward, at least the way I'm thinking about it. I eventually managed to get the button to execute an Automation, which can get a bash script started which at least gets as far as creating the queries

```
for l in $(head -c $(expr 10 \* 6) /dev/urandom | base64 --wrap=8 | tr '[:lower:]' '[:upper:]' | sed 's/[\+\/=]/0/g'); do
		echo "INSERT INTO activation_tokens(token, uses, max_uses) VALUES ('$l', 0, 1);"
done
```

(still didn't get as far as how I'd pass the parameter of the number of tokens to generate.) But there is no way to actually load this query into the database, so this seems pointless. (And also, I'd assume the automation runs in the background even if triggered by a user action - which is not exactly desired in this case).

What would be the best way to try and achieve this? My next thought would be to use my api backend as a connector and use it to actually perform this thing. But that doesn't seem right.

**Comment by xnohat** [comment]
My idea is you using bash to generate token -> send token to External Data Connector step (using binding and query params) -> write to database
